src/components/atoms/MonacoEditor/MonacoEditor.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {lazy, useEffect, useMemo, useState} from 'react';
+import type {EditorDidMount, monaco} from 'react-monaco-editor';
+
+import Colors from '@styles/Colors';
+
+const MonacoEditor = lazy(() => import('react-monaco-editor'));
+
+type TkMonacoEditorProps = {
+  language: string;
+  height?: string;
+  minHeight?: number;
+  value: string;
+  onChange: (value: string) => void;
+};
+
+export const lineHeight = 22;
+
+const options = {
+  contextmenu: true,
+  fontFamily: 'Roboto Mono, Monaco, monospace',
+  fontSize: 13,
+  lineHeight,
+  minimap: {
+    enabled: false,
+  },
+  scrollbar: {
+    horizontalSliderSize: 4,
+    verticalSliderSize: 18,
+  },
+  selectOnLineNumbers: true,
+  readOnly: false,
+  automaticLayout: true,
+  padding: {
+    top: 12,
+    bottom: 12,
+  },
+};
+
+const defaultEditorHeight = 300;
+
+const TkMonacoEditor: React.FC<TkMonacoEditorProps> = props => {
+  const {value, onChange, language, height, minHeight = defaultEditorHeight} = props;
+  const [editor, setEditor] = useState<monaco.editor.IStandaloneCodeEditor | null>(null);
+
+  const handleEditorDidMount: EditorDidMount = (mountedEditor, monaco) => {
+    monaco.editor.defineTheme('testkube-theme', {
+      base: 'vs-dark',
+      inherit: true,
+      rules: [
+        {token: 'type', foreground: Colors.amber400},
+        {token: 'delimiter', foreground: Colors.slate50},
+        {token: 'string.yaml', foreground: Colors.lime400},
+        {token: 'number', foreground: Colors.rose400},
+      ],
+      colors: {
+        'editor.background': Colors.slate800,
+        'editor.selectionBackground': Colors.indigo400,
+        'editor.lineHighlightBackground': Colors.slate600,
+      },
+    });
+    setEditor(mountedEditor);
+  };
+
+  const editorMaxHeight = useMemo(() => {
+    return window.innerHeight * 0.5;
+  }, [window.innerHeight]);
```

Comment:
`useMemo` is not necessary there, as it anyway needs to read `window.innerHeight` in that implementation.

Besides, it could be more reliable using [`useWindowSize`](https://github.com/streamich/react-use/blob/master/src/useWindowSize.ts):

```suggestion
  const editorMaxHeight = useWindowSize().height * 0.5;
```

Reply:
Got it, makes sense